example/namedtracer/foo/foo.go

```diff
@@ -28,16 +31,23 @@ var (
 
 // SubOperation is an example to demonstrate the use of named tracer.
 // It creates a named tracer with its package path.
-func SubOperation(ctx context.Context) error {
+func SubOperation(ctx context.Context, logger log.Logger) error {
 	// Using global provider. Alternative is to have application provide a getter
 	// for its component to get the instance of the provider.
 	tr := otel.Tracer("example/namedtracer/foo")
 
 	var span trace.Span
-	_, span = tr.Start(ctx, "Sub operation...")
+	ctx, span = tr.Start(ctx, "Sub operation...")
 	defer span.End()
 	span.SetAttributes(lemonsKey.String("five"))
 	span.AddEvent("Sub span event")
 
+	// Log via Otel Logger API
+	logger.Emit(ctx, log.WithAttributes(attribute.String("operation", "suboperation")))
+
+	// Log via slog API. Note this will correctly output Span Context (traceid,spanid) in the final output
+	// if Otel slog Handler is used.
+	slog.InfoCtx(ctx, "Otel logs via slog and with context!", "source", "slog")
```

Comment:
@MrAlias See how this correctly does Otel context-aware logging although it is totally unaware of what Otel is!